gpt_index/readers/chroma.py

```diff
@@ -0,0 +1,71 @@
+"""Chroma Reader."""
+
+from typing import Any, Optional
+
+from gpt_index.readers.base import BaseReader
+from gpt_index.readers.schema.base import Document
+
+
+class ChromaReader(BaseReader):
+    """Chroma reader.
+
+    Retrieve documents from existing persisted Chroma collections.
+
+    Args:
+        collection_name: Name of the peristed collection.
+        persist_directory: Directory where the collection is persisted.
+
+    """
+
+    def __init__(
+        self,
+        collection_name: Optional[str] = None,
+        persist_directory: Optional[str] = None,
+    ) -> None:
+        """Initialize with parameters."""
+        import_err_msg = (
+            "`chromadb` package not found, please run `pip install chromadb`"
+        )
+        try:
+            import chromadb  # noqa: F401
+        except ImportError:
+            raise ValueError(import_err_msg)
+
+        if (collection_name is None) or (persist_directory is None):
```

Comment:
if this is the case why not just make `collection_name` and `persist_directory` required args and not optional? 

for some context the main reason we do that in `BaseGPTIndex` subclasses is due to the fact that we initialize a bunch of defaults if the provided args are None (and tbf it's not the cleanest interface). In the reader class, if we can, let's use required args.